test/unit/test_il_vm.c

```diff
@@ -143,28 +143,27 @@ static bool test_rzil_vm_root_evaluation() {
 	//	true
 	//	false)
 	// evaluate (add 23 19) will get a bitvector, but condition require a bool
-	RzILOp *arg1 = rz_il_op_new_bitv_from_st64(16, 23);
-	RzILOp *arg2 = rz_il_op_new_bitv_from_st64(16, 19);
-	RzILOp *add = rz_il_op_new_add(arg1, arg2);
-	RzILOp *true_val = rz_il_op_new_b1();
-	RzILOp *false_val = rz_il_op_new_b0();
-	RzILOp *ite_root = rz_il_op_new_ite(add, true_val, false_val);
+	RzILOpBitVector *arg1 = rz_il_op_new_bitv_from_st64(16, 23);
+	RzILOpBitVector *arg2 = rz_il_op_new_bitv_from_st64(16, 19);
+	RzILOpBitVector *add = rz_il_op_new_add(arg1, arg2);
+	RzILOpBool *condition = rz_il_op_new_bool_inv(rz_il_op_new_eq(add, rz_il_op_new_bitv_from_ut64(16, 0)));
```

Comment:
so now bu
itv are not automatically converted to bool ?

Reply:
Correct. Bitvectors and bool are two different things and BAP does not do this either. This particular example is a bit overcomplicated for testing, usually you will just want to use `rz_il_op_new_non_zero()`.

Reply:
this is kinda a problem for me, but i do understand the point.